test/spec/oauthUtil.js

```diff
@@ -601,10 +606,139 @@ describe('validateClaims', function () {
 
   it('throws an AuthSdkError when no issuer is provided', function () {
     var fn = function () {
-      oauthUtil.validateClaims(sdk, undefined, {
+      oauthUtil.validateClaims(sdk, {}, {
         clientId: 'foo'
       });
     };
     expect(fn).toThrowError('The jwt, iss, and aud arguments are all required');
   });
+
+  it('validates nonce, if provided', function() {
+    validationOptions.nonce = 'bar';
+    var claims = { nonce: 'foo' };
+    var fn = function () {
+      oauthUtil.validateClaims(sdk, claims, validationOptions);
+    };
+    expect(fn).toThrowError('OAuth flow response nonce doesn\'t match request nonce');  
+  });
+
+  it('validates issuer', function() {
+    var claims = { iss: 'foo' };
+    var fn = function () {
+      oauthUtil.validateClaims(sdk, claims, validationOptions);
+    };
+    expect(fn).toThrowError('The issuer [' + claims.iss + '] ' +
+    'does not match [' + validationOptions.issuer + ']'); 
+  });
+
+  it('validates audience', function() {
+    var claims = {
+      iss: validationOptions.issuer,
+      aud: 'nobody'
+    };
+    var fn = function () {
+      oauthUtil.validateClaims(sdk, claims, validationOptions);
+    };
+    expect(fn).toThrowError('The audience [' + claims.aud + '] ' +
+      'does not match [' + validationOptions.clientId + ']'); 
+  });
+
+  it('validates exp > iat', function() {
+    var claims = {
+      iss: validationOptions.issuer,
+      aud: validationOptions.clientId,
+      exp: 1,
+      iat: 2
+    };
+    var fn = function () {
+      oauthUtil.validateClaims(sdk, claims, validationOptions);
+    };
+    expect(fn).toThrowError('The JWT expired before it was issued'); 
+  });
+
+  it('throws if expired', function() {
+    var now = 10;
+    util.warpToUnixTime(now);
+    var claims = {
+      iss: validationOptions.issuer,
+      aud: validationOptions.clientId,
+      exp: now - 1,
+      iat: now - 2
+    };
+    sdk.options.maxClockSkew = 0;
+    var fn = function () {
+      oauthUtil.validateClaims(sdk, claims, validationOptions);
+    };
+    expect(fn).toThrowError('The JWT expired and is no longer valid'); 
+  });
+
+  it('maxClockSkew extends expiration window', function() {
+    var now = 10;
+    var skew = 2;
+    util.warpToUnixTime(now);
+    var claims = {
+      iss: validationOptions.issuer,
+      aud: validationOptions.clientId,
+      exp: now - 1,
+      iat: now - 2
+    };
+    sdk.options.maxClockSkew = skew;
+    var fn = function () {
+      oauthUtil.validateClaims(sdk, claims, validationOptions);
+    };
+    expect(fn).not.toThrowError();
+    util.warpToUnixTime(now + skew);
+    expect(fn).toThrowError('The JWT expired and is no longer valid'); 
```

Comment:
The first validation (with now = 10) would normally fail (without skew) because exp = 9. It only passes because of the skew which makes calculated exp = 11. That's the positive test for skew logic. The next test (with now = 12) is the expected failure beyond the skew window

Reply:
Reading it again, it makes sense now ☕️